doc/database/multi_database_instances.md

```diff
@@ -820,36 +820,159 @@ EXEC_WITH_ERROR_THROW(cmd, res);
+- [x] In new image, there are only three instances and after startup, each instances has full data. But ins0 only use DB0&DB1 based on configration in database_config.json, DB2&DB3 are never used and we can flushdb them.
+
+![center](./img/db_restore_new.png)
 
-1. we need to only allow warm-reboot/upgrade from one version to another version. In the new image, we can handle the warm-reboot/upgrading accordingly and know where to migrate the database since the "delta" between old and new database_config.json can be calculated. We can write a script to handle this case.
-2. other idea is welcome
+Now we see, the extra step for the new implementation is migrating all data into one redis instance and the "flushdb" operations after new redis instances started.
+
+- For "flushdb" operations, it takes ~0.3 second. If we want to save this time as well , we can delay this operations until warmboot is done, otherwise we can add these operations during warmboot.
+
+- For the data migration, I did some test on local.
+  
+   The CPU on my local setup is as below, better CPU may have better performance:
+
+  ```shell
+  admin@ASW-7005:~$ lscpu
+  Architecture:        x86_64
+  CPU op-mode(s):      32-bit, 64-bit
+  Byte Order:          Little Endian
+  Address sizes:       36 bits physical, 48 bits virtual
+  CPU(s):              4
+  On-line CPU(s) list: 0-3
+  Thread(s) per core:  1
+  Core(s) per socket:  4
+  Socket(s):           1
+  NUMA node(s):        1
+  Vendor ID:           GenuineIntel
+  CPU family:          6
+  Model:               77
+  Model name:          Intel(R) Atom(TM) CPU  C2558  @ 2.40GHz
+  Stepping:            8
+  CPU MHz:             2393.975
+  CPU max MHz:         2400.0000
+  CPU min MHz:         1200.0000
+  BogoMIPS:            4787.94
+  Virtualization:      VT-x
+  L1d cache:           24K
+  L1i cache:           32K
+  L2 cache:            1024K
+  NUMA node0 CPU(s):   0-3
+  ```
+  
+  - [x] Third part script like redis-dump/load is very slow, usually takes **~23s** when data size is **~40K**
+  - [x] redis-cli cmd shown as below works better, takes **~3s** when data size is **~40K**
+  
+  ```shell
+  redis-cli -n 3 --raw KEYS '*' | xargs redis-cli -n 3 MIGRATE 127.0.0.1 6380 "" 1 5000 KEYS
+  ```
+
+  - [x] I also tried lua script as below, this way is the best, it takes about **~1s **when data size is **~40K** and **~2s** when data size is **~100K**.  Sample codes migratedb as below:
+
+  ```python
+  #!/usr/bin/python
+  from __future__ import print_function
+  import sys
+  import swsssdk
+  import redis
+  
+  dblists = swsssdk.SonicDBConfig.get_dblist()
+  for dbname in dblists:
+      dbsocket = swsssdk.SonicDBConfig.get_socket(dbname)
+      #dbport = swsssdk.SonicDBConfig.get_port(dbname)
+      dbid = swsssdk.SonicDBConfig.get_dbid(dbname)
+      dbhost = swsssdk.SonicDBConfig.get_hostname(dbname)
+
+      r = redis.Redis(host=dbhost, unix_socket_path=dbsocket, db=dbid)
```

Comment:
Why mix TCP socket and unix socket?

Reply:
it is not mix, if unix_socket_path is there, it always use unix socket, the host and port parameters are ignored. I leave host there since sometimes I changed  unix_socket_path to port for switching mode.